app/team.go

```diff
@@ -88,11 +88,28 @@ func (a *App) isTeamEmailAddressAllowed(email string, allowedDomains string) boo
 	return true
 }
 
+func (a *App) isGuestEmailAddressAllowed(email string) bool {
```

Comment:
This new method has a lot of overlap with the old `isTeamEmailAddressAllowed` method. Why don't we rename the old method to `isEmailAddressAllowed` and pass `*model.User` to it, and depending on whether the user is guest or not, perform the appropriate logic. That way, there is less duplication and in future, one doesn't have to remember that there are 2 different functions depending on whether a user is guest or not.

Reply:
Yes I know. I though about the two approaches and finally I've taken mainly for two reasons:

- Having all the logic in the same method looks a bit confusing. We have to create a variable depending on the user.isGuest, getting different configuration and so on.
- Even though right now the code is duplicated we can leverage that in the future is that changes. A bit of duplication is not hard

But I've found a better way to do a mix of the proposed solutions :)

Reply:
Yes. I though a bit about how to handle it.

I dismissed to include everything into the same method because it seems to broke the single responsibility a bit too much: allowed domains logic, email normalization logic, domain belonging logic...

But you're right, my decision wasn't better so I've came up with a much better approach I hope you like it as well :)

Reply:
Looks great now.